src/asynqp/queue.py

```diff
@@ -317,3 +386,177 @@ def error(self, exc):
         for consumer in self.consumers.values():
             if hasattr(consumer.callback, 'on_error'):
                 consumer.callback.on_error(exc)
+
+
+class _ConsumerContext(_UserCoroutine):
+    """ Coroutine proxy, that can be used as async context manager
+    """
+
+    if PY_35:
+
+        @asyncio.coroutine
+        def __aenter__(self):
+            self._consumer = yield from self._coro
+            return self._consumer
+
+        @asyncio.coroutine
+        def __aexit__(self, exc_type, exc, tb):
+            yield from self._consumer.cancel()
+
+
+class QueuedConsumer:
```

Comment:
I wonder if it might make sense to make this class _inherit_ `asyncio.Queue`, rather than just wrapping it. Then you'd get all of `Queue`'s methods 'for free' (in particular I think `get_nowait` might be useful). What do you think?